src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -308,6 +308,40 @@ public void Tessellate(IRenderPackage package, TessellationParameters parameters
             // or transactions and which must necessarily be threaded in a specific way.
         }
 
+        /// <summary>
+        /// Delete the element from the document. 
+        /// </summary>
+        /// <returns>true if successfully deleted, false otherwise.</returns>
+        public bool Delete()
+        {
+            // Document to work with
+            Autodesk.Revit.DB.Document document = this.InternalElement.Document;
+            
+            // Start the transaction
+            TransactionManager.Instance.EnsureInTransaction(document);
+
+            // Delete the element
+            try
+            {
+                document.Delete(this.InternalElementId);
+            }
+            catch(Autodesk.Revit.Exceptions.ArgumentException)
```

Comment:
is it worth also rethrowing these exceptions so users see the exception messages on the nodes?

Reply:
@mjkkirschner Can you give an example, or it is as simple as setting the node warning?

Reply:
https://github.com/DynamoDS/DynamoRevit/blob/master/src/Libraries/RevitNodes/Elements/DirectShape.cs#L369

see this example for directshape - any exception that gets throw (in general) will be used to set the warning state on a node - it will turn yellow and the error popup will display the exception type and inner message.

Reply:
@mjkkirschner OK, so for example, if I just let the exceptions propagate out? (i.e. I don;'t need to catch.. I just need to add finally for transaction work for example, but if the exceptions occur, they will propagate out?

Reply:
on second thought - let me show another example - it seems to me you want to return a valid output - this can also be done while still putting the node into a warning state -with a different API. one sec.

Reply:
here is an example of using the `OnLogWarningMessage` method to put the node into a warning state while still returning a valid output:

https://github.com/DynamoDS/Dynamo/blob/ec10f936824152e7dd7d6d019efdcda0d78a5264/src/Libraries/DSOffice/Excel.cs#L427

not required, just something that is possible if you want both. This method is part of the DynamoServices.dll which DynamoRevit is likely already referencing.

Reply:
when you throw an exception from a zero touch node,  it will propagate up to Dynamo and get caught, and the node will go into an error state and return `null` - if you want to return a valid output - you don't want to throw, and can continue the pattern you are using :).

If you also want to let the user know why the element was not deleted you can set the node in an error state but still return your valid output of`false`using that logWarning method.

Reply:
@mjkkirschner thank you. In hindsight, the pattern of returning an empty list in this probably not ideal in this case. The delete should delete and if it doesn't it is best to throw rather than have the script continue along without having done so, regardless of an empty list being returned etc.